### PR TITLE
Update to go1.20

### PR DESCRIPTION
**Pull request by stlava**
The goal of this PR is to primarily update to go1.20 and update any necessary code.

Summery of changes:
- Updates to go 1.20
- Cleans up circleci build and removes publishing binaries. Moving forward we'll only publish docker images.
- Updates integration tests to run against postgres 14.7
- Fixes race condition in integration tests.
- Switches from `master` to `main` branch
- Updates `localstack` to 2.x and update ports to match new localstack settings
- Adds some more debugging and fixes up the itest datapoller.
- Extended time of integration tests and hardware profile as it seems there may have been a performance regression on the circleci side.

### CODE REVIEW

File: itests/tests/kinesis/test_multi_worker_no_kinesis_errors/envfile.env [hunk 1]
```diff
@@ -1,4 +1,5 @@
 EXPECTED_COUNT=50000
 WORKERS=4
 MEMPROFILE=/perf/mem.pprof
-CPUPROFILE=/perf/cpu.pprof
\ No newline at end of file
+CPUPROFILE=/perf/cpu.pprof
+KINESIS_POLLER_WAIT_TIME=180
```
Comment (ruban-rengaraju) [inline itests/tests/kinesis/test_multi_worker_no_kinesis_errors/envfile.env:5]: nit: newline

File: Makefile [hunk 2]
```diff
@@ -40,7 +40,7 @@ clean:
 build: generate
 	@echo "Creating GO binary"
 	mkdir -p target
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "$(GO_LDFLAGS)" -o target/pg-bifrost github.com/Nextdoor/pg-bifrost.git/main
+	CGO_ENABLED=0 GOOS=linux go build -ldflags "$(GO_LDFLAGS)" -o target/pg-bifrost github.com/Nextdoor/pg-bifrost.git/main
```
Comment (ruban-rengaraju) [inline Makefile:43]: this supports multi arch build?

Comment (stlava) [reply Makefile:43]: Yes but we don't produce multi arch docker images right now.

**Review by ruban-rengaraju — APPROVED** [review]
overall looks good, left a few non-blocking comments.

**Pull request merged**